src/AccentColor/AccentColorManager.vala

```diff
@@ -0,0 +1,197 @@
+
+    private Gala.AccountsService? gala_accounts_service = null;
+
+    private Settings background_settings;
+    private Settings interface_settings;
+
+    private NamedColor[] theme_colors = {
+        new NamedColor () {
+            name = "Blue",
+            theme = "blueberry",
+            hex = "#3689e6"
+        },
+        new NamedColor () {
+            name = "Mint",
+            theme = "mint",
+            hex = "#28bca3"
+        },
+        new NamedColor () {
+            name = "Green",
+            theme = "lime",
+            hex = "#68b723"
+        },
+        new NamedColor () {
+            name = "Yellow",
+            theme = "banana",
+            hex = "#f9c440"
+        },
+        new NamedColor () {
+            name = "Orange",
+            theme = "orange",
+            hex = "#ffa154"
+        },
+        new NamedColor () {
+            name = "Red",
+            theme = "strawberry",
+            hex = "#ed5353"
+        },
+        new NamedColor () {
+            name = "Pink",
+            theme = "bubblegum",
+            hex = "#de3e80"
+        },
+        new NamedColor () {
+            name = "Purple",
+            theme = "grape",
+            hex = "#a56de2"
+        },
+        new NamedColor () {
+            name = "Brown",
+            theme = "cocoa",
+            hex = "#8a715e"
+        },
+        new NamedColor () {
+            name = "Gray",
+            theme = "slate",
+            hex = "#667885"
+        }
+    };
+
+    construct {
+        background_settings = new Settings ("org.gnome.desktop.background");
+        interface_settings = new Settings (INTERFACE_SCHEMA);
+
+        string? user_path = null;
+        try {
+            FDO.Accounts? accounts_service = GLib.Bus.get_proxy_sync (
+                GLib.BusType.SYSTEM,
+               "org.freedesktop.Accounts",
+               "/org/freedesktop/Accounts"
+            );
+
+            user_path = accounts_service.find_user_by_name (GLib.Environment.get_user_name ());
+        } catch (Error e) {
+            critical (e.message);
+        }
+
+        if (user_path != null) {
+            try {
+                gala_accounts_service = GLib.Bus.get_proxy_sync (
+                    GLib.BusType.SYSTEM,
+                    "org.freedesktop.Accounts",
+                    user_path
+                );
+
+                ((DBusProxy)gala_accounts_service).g_properties_changed.connect (() => {
+                    update_accent_color ();
+                });
+            } catch (Error e) {
+                warning ("Unable to get AccountsService proxy, accent color preference may be incorrect");
+            }
+        }
+
+        background_settings.changed["picture-uri"].connect (update_accent_color);
+
+        update_accent_color ();
+    }
+
+    private void update_accent_color () {
+        bool set_accent_color_based_on_wallpaper = gala_accounts_service.prefers_accent_color == 0;
```

Comment:
Most likely because I didn't installed https://github.com/elementary/switchboard-plug-pantheon-shell/pull/263 due to outdated Granite, but it changes my accent color when I restart Gala.

Reply:
You can test elementary/switchboard-plug-pantheon-shell#263 if you also install https://github.com/elementary/granite/pull/483